blockchain/blockchain.go

```diff
@@ -336,47 +336,50 @@ func (b *Blockchain) Store(block *core.Block, blockCommitments *core.BlockCommit
 	stateUpdate *core.StateUpdate, newClasses map[felt.Felt]core.Class,
 ) error {
 	return b.database.Update(func(txn db.Transaction) error {
-		if err := verifyBlock(txn, block); err != nil {
+		if err := b.storeBlock(txn, block, blockCommitments); err != nil {
 			return err
 		}
 
-		state := core.NewState(txn)
-		if err := state.Update(block.Number, stateUpdate.StateDiff, newClasses); err != nil {
+		if err := core.NewState(txn).Update(block.Number, stateUpdate.StateDiff, newClasses); err != nil {
 			return err
 		}
 
-		if err := b.verifyStateUpdateRoot(state, stateUpdate.NewRoot); err != nil {
+		if err := b.verifyStateUpdateRoot(core.NewState(txn), stateUpdate.NewRoot); err != nil {
```

Comment:
Is there a reason to use two different state instances?

Reply:
No real reason. Since `core.State` is just a wrapper around a `db.Transaction` it is safe to assume all changes are persisted in the `txn`. But will change it to use a single state instance to make easier to follow.